.github/workflows/release.yml

```diff
@@ -0,0 +1,40 @@
+name: kapp-controller release
+
+on:
+  push:
+    tags:
+      - 'v*'
+
+jobs:
+  kapp-controller-release:
+    name: kapp-controller release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Check out code into the Go module directory
+        uses: actions/checkout@v1
+        with:
+          path: src/github.com/${{ github.repository }}
+      - name: Install Carvel Tools
+        uses: vmware-tanzu/carvel-setup-action@v1
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+          only: ytt, kbld
+          ytt: v0.35.1
+          kbld: v0.30.0
+      - name: Run release script
+        env:
+          REGISTRY_PASSWORD: ${{ secrets.REGISTRY_PASSWORD }}
+        run: |
+          set -e -x
+
+          # TODO: Update login process once new registry
+          # is chosen. See https://github.com/vmware-tanzu/carvel-kapp-controller/issues/319.
+          docker login -u k14s -p "$REGISTRY_PASSWORD"
```

Comment:
Making quick note here that ghcr.io might have a way to support not storing registry credentials in GitHub. So this docker login might be unnecessary in final implementation.